util_test.go

```diff
@@ -27,3 +28,80 @@ func TestFileExistsReturnsFalseForNonExistingFiles(t *testing.T) {
 	assertEqual(t, fileExists(("util_nope.go")), false)
 	assertEqual(t, fileExists(("util_nope_test.go")), false)
 }
+
+func TestDefaultReleaseSentryReleaseEnvvar(t *testing.T) {
+	releaseVersion := "1.2.3"
+	cleanup := envSetter(map[string]string{
+		"SENTRY_RELEASE": releaseVersion,
+	})
+
+	assertEqual(t, defaultRelease(), releaseVersion)
+
+	t.Cleanup(cleanup)
+}
+
+func TestDefaultReleaseSentryReleaseEnvvarPrecedence(t *testing.T) {
+	releaseVersion := "1.2.3"
+	cleanup := envSetter(map[string]string{
+		"SOURCE_VERSION": "3.2.1",
+		"SENTRY_RELEASE": releaseVersion,
+	})
+
+	assertEqual(t, defaultRelease(), releaseVersion)
+
+	t.Cleanup(cleanup)
+}
+
+func TestRevisionFromBuildInfo(t *testing.T) {
+	releaseVersion := "deadbeef"
+
+	info := &debug.BuildInfo{
+		Main: debug.Module{
+			Path:    "my/module",
+			Version: "(devel)",
+		},
+		Deps: []*debug.Module{
+			{
+				Path:    "github.com/getsentry/sentry-go",
+				Version: "v0.23.1",
+				Replace: &debug.Module{
+					Path: "pkg/sentry",
+				},
+			},
+		},
+		Settings: []debug.BuildSetting{
+			{
+				Key:   "vcs.revision",
+				Value: releaseVersion,
+			},
+		},
+	}
+	cleanup := envSetter(map[string]string{})
```

Comment:
This line won't be necessary, right?

Reply:
This line was added to ensure that no environment variables were accidentally set (locally or in CI) which could cause the release value to be different than the info provided in `debug.BuildInfo`